compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -268,6 +268,9 @@ impl<'a> StringReader<'a> {
                 // tokens like `<<` from `rustc_lexer`, and then add fancier error recovery to it,
                 // as there will be less overall work to do this way.
                 let token = unicode_chars::check_for_substitution(self, start, c, &mut err);
+                if c == '\x00' {
+                    err.help("source files must be encoded in UTF-8, unexpected null bytes might occur when the wrong text encoding is used");
```

Comment:
Wording nit: "The wrong text encoding" should read as "a wrong text encoding" or maybe "a different text encoding".

How about:

> source files must contain UTF-8 encoded text, unexpected null bytes might occur when a different encoding is used

Reply:
Heh, pickfire was first.